### PR TITLE
mysql-sandbox 3.1.0

### PR DESCRIPTION
**Pull request by denji**
https://github.com/datacharmer/mysql-sandbox/releases/tag/3.1.00

### CODE REVIEW

**Comment by dunn** [comment]
The build is getting killed immediately by the sandbox:
```
==> Downloading https://github.com/datacharmer/mysql-sandbox/archive/3.1.00.tar.gz
Already downloaded: /Library/Caches/Homebrew/mysql-sandbox-3.1.00.tar.gz
==> Verifying mysql-sandbox-3.1.00.tar.gz checksum
tar xf /Library/Caches/Homebrew/mysql-sandbox-3.1.00.tar.gz
Error: Operation not permitted - /usr/local/lib
```

**Comment by DomT4** [comment]
It's this line:

```ruby
 ENV.prepend_create_path "PERL5LIB", "#{HOMEBREW_PREFIX}/lib/perl5/site_perl"
```

It needs to become either `lib/"perl5/site_perl"` or `libexec/"perl5/site_perl"`. I'd probably recommend the latter unless other software is known to need to hook into the Perl elements of the formula, and then doing something like [this](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/dpkg.rb#L41-L43).

File: Library/Formula/mysql-sandbox.rb [hunk 1]
```diff
@@ -11,7 +12,7 @@ class MysqlSandbox < Formula
   end
 
   def install
-    ENV.prepend_create_path "PERL5LIB", "#{HOMEBREW_PREFIX}/lib/perl5/site_perl"
+    ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5/site_perl/"
 
     system "perl", "Makefile.PL", "PREFIX=#{prefix}"
```
Comment (DomT4) [inline Library/Formula/mysql-sandbox.rb]: If we're going with `libexec` this should be changed to `libexec` as well so everything is still in the same directory, and then [this logic](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/dpkg.rb#L41-L43) used to put the executables back in the top-level directory which'll be symlinked into the `$PATH`.

**Comment by bfontaine** [comment]
Ping?

**Comment by denji** [comment]
I applied all the changes, I hope it won't break the build.

**Comment by dunn** [comment]
Merged in cfaf42d0127e226b0eb5a33fac427bbec3bcbf97, thanks and sorry for the delay!

**Comment by AlessandroBerardi** [comment]
Hi, I'm having an issue installing the latest mysql-sandbox via brew, and it seems to be related to the changes in this pull request:

```
brew install mysql-sandbox
# Installs with no errors

make_sandbox
# Returns the following error:
Can't locate MySQL/Sandbox.pm in @INC (you may need to install the MySQL::Sandbox module) (@INC contains: /usr/local/Cellar/mysql-sandbox/3.1.00/libexec/lib/perl5 /Library/Perl/5.18/darwin-thread-multi-2level /Library/Perl/5.18 /Network/Library/Perl/5.18/darwin-thread-multi-2level /Network/Library/Perl/5.18 /Library/Perl/Updates/5.18.2/darwin-thread-multi-2level /Library/Perl/Updates/5.18.2 /System/Library/Perl/5.18/darwin-thread-multi-2level /System/Library/Perl/5.18 /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level /System/Library/Perl/Extras/5.18 .) at /usr/local/Cellar/mysql-sandbox/3.1.00/libexec/bin/make_sandbox line 25.
BEGIN failed--compilation aborted at /usr/local/Cellar/mysql-sandbox/3.1.00/libexec/bin/make_sandbox line 25.
```

**Comment by dunn** [comment]
Could you open a new issue?  Make sure to follow the [Troubleshooting guidelines](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting) too.  Thanks!

**Comment by DomT4** [comment]
@AlessandroBerardi https://github.com/Homebrew/homebrew/pull/48443

**Comment by AlessandroBerardi** [comment]
I apologiste for not following the guidelines @dunn.

Thanks for the fix @DomT4!

**Pull request closed without merging**